src/main/java/org/apache/rocketmq/flink/legacy/RocketMQSourceFunction.java

```diff
@@ -404,8 +380,7 @@ private long getMessageQueueOffset(MessageQueue mq) throws MQClientException {
     private void updateMessageQueueOffset(MessageQueue mq, long offset) throws MQClientException {
         offsetTable.put(mq, offset);
         if (!enableCheckpoint) {
-            consumer.updateConsumeOffset(mq, offset);
-            consumer.getOffsetStore().persist(consumer.queueWithNamespace(mq));
+            consumer.getOffsetStore().updateOffset(mq, offset, false);
```

Comment:
Why not persist the offset? IMO, the offset should be persisted here. cc @ShannonDing @zhouxinyu

Reply:
It depends. The consumer will persist the offsets periodically, even if we only call `updateOffset`. The `persist` call is heavy, each call will trigger a RPC.

So, if the updateMessageQueueOffset already batched some requests, `persist` is more suitable. If not and we can tolerate redundant messages in some abnormal cases, updateOffset is enough.